pylint/checkers/variables.py

```diff
@@ -2378,6 +2375,13 @@ def _is_never_evaluated(
                 return True
         return False
 
+    @staticmethod
+    def _is_variable_annotation_in_function(node: nodes.NodeNG) -> bool:
+        is_annotation = utils.get_node_first_ancestor_of_type(node, nodes.AnnAssign)
+        return is_annotation and utils.get_node_first_ancestor_of_type(
```

Comment:
```suggestion
        return is_annotation and utils.get_node_first_ancestor_of_type( # type: ignore[return-value]
```
astroid does not declare its types, but it's going to be a boolean not Any.

Reply:
Thanks for the suggestion!